src/scalar.h

```diff
@@ -19,6 +19,34 @@
 #error "Please select wide multiplication implementation"
 #endif
 
+#ifndef VERIFY
+/* In non-VERIFY mode, we #define the scalar operations to be identical to their
+ * internal scalar implementation, to avoid the potential overhead of a
+ * function call (even though presumably inlinable). */
+#  define secp256k1_scalar_clear secp256k1_scalar_impl_clear
```

Comment:
Hm, when I opened #1360 I didn't except that we want this indirection also here, but I think it makes sense.

On the one hand, the situation here is similar to the field code in the sense that we have multiple scalar implementations. On the other hand, the invariant here (no overflow) is inherently a property of the implementation, and does not leak into the scalar interface. So, you could argue that checking this is a concern of the implementation. But hm, what you do here is simply introduce a generic way for implementations to provide a verify function (which happens to be very similar here in all three implementations).  And this makes sense.

To be clear: What you propose here makes sense and solves the problem, I'm just wrapping my around whether it's the cleanest/most elegant solution. cc @sipa who worked on this in the field code.